src/util/Iterators.h

```diff
@@ -408,4 +408,13 @@ class InputRangeTypeErased {
   decltype(auto) end() { return impl_->end(); }
   using iterator = typename InputRangeFromGet<ValueType>::Iterator;
 };
+
+template <typename Range>
+auto getSingleElement(Range g) {
```

Comment:
Okay, here we need to avoid accidental copies.
Either we have to constrain this to be a ` view`  and then work with `views::all`  which has some implications on several compilers, 
or you should just take the range by `Range&&`. As this is a synchronous function, there can be no lifetime issues.

And as you always move the element, maybe call it `getMovedSingleElement` or something, and please add a comment.